src/dbnode/storage/index.go

```diff
@@ -194,7 +208,8 @@ func newNamespaceIndexWithOptions(
 	idx := &nsIndex{
 		state: nsIndexState{
 			runtimeOpts: nsIndexRuntimeOptions{
-				insertMode: indexOpts.InsertMode(), // FOLLOWUP(prateek): wire to allow this to be tweaked at runtime
+				insertMode:            newIndexOpts.opts.IndexOptions().InsertMode(), // FOLLOWUP(prateek): wire to allow this to be tweaked at runtime
+				flushBlockNumSegments: runtime.DefaultFlushIndexBlockNumSegments,
```

Comment:
This can now be `runtimeOptsMgr.FlushIndexBlockNumSegments()` rather than referencing the default.